### PR TITLE
feat(Accept financial contributions): Add organization flow

### PR DESCRIPTION
**Pull request by sbinlondon**
This PR add the organization flow.

I've chosen to make 'myself' and 'organization' flows into one component because the two separate components would have been so similar in themselves, it makes sense to me to have one single one with conditional rendering. I do realise however that this is a huge component, so I've tried to comment and make things as clear as possible.

This hopefully also clarifiies some of the conditional rendering that was happening in `StripeOrBankAccount` which was preparing for the case where either a host organization or the `LoggedInUser` as host was being passed into that component based on whether it is the 'myself' or 'organization' flow.

With this PR you should now be able to test all combinations of possiblities:

* choose 'myself', 'organization', or 'apply to host' at the beginning in the `ContributionCategoryPicker`
* if you choose 'organization', you should be able to to pick your org or create one
* then for 'myself' or 'organization' you picked between connecting Stripe or a bank account...
* ...unless either  Stripe or bank account are already connected, or if both are, in which case it will show that

### CODE REVIEW

File: components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js [hunk 1]
```diff
@@ -100,8 +139,21 @@ class AcceptContributionsMyself extends React.Component {
 
   render() {
     const { collective, router, LoggedInUser } = this.props;
-    const { error } = this.state;
+    const { error, miniForm, organization, loading } = this.state;
 
+    // Get and filter orgs LoggedInUser is part of
+    const memberships = uniqBy(
+      LoggedInUser.memberOf.filter(m => m.role !== 'BACKER'),
```
Comment (znarf) [inline components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js:146]: Should that be `m.role === 'ADMIN'`?

**Comment by znarf** [comment]
As already discussed, we should have the Organization Mini Form without extra "Admin email address" and "Admin name", the admin should be current authenticated user.

**Comment by sbinlondon** [comment]
> As already discussed, we should have the Organization Mini Form without extra "Admin email address" and "Admin name", the admin should be current authenticated user.

Agreed! Should I make some changes to the original component and maybe pass it as a prop, like `noAdminFields`?

**Comment by znarf** [comment]
Capture d’écran 2020-04-24 à 13 49 16

To not forget, "Personal Account" is hardcoded and will be inaccurate. At this stage, it doesn't make sense, maybe we can bring it back when we'll have the upgraded Bank Account form.

**Comment by sbinlondon** [comment]
> Capture d’écran 2020-04-24 à 13 49 16
> 
> To not forget, "Personal Account" is hardcoded and will be inaccurate. At this stage, it doesn't make sense, maybe we can bring it back when we'll have the upgraded Bank Account form.

Shall we just have 'Stripe connected' and 'Bank account set up' with the checkmark, nothing else?

**Comment by sbinlondon** [comment]
Mini form and Stripe/bank account picker updated

Screenshot 2020-04-24 at 13 21 27

Screenshot 2020-04-24 at 13 20 19

**Comment by znarf** [comment]
> Shall we just have 'Stripe connected' and 'Bank account set up' with the checkmark, nothing else?

I'm good with that.

**Comment by sbinlondon** [comment]
Things to review on Monday:

* changes made to the `CreateCollectiveMIniForm` to allow you to create an org without displaying the admin name and email fields
* changes to the `StripeOrBankAccountPicker` copy as discussed on Slack
* Copy changes on the success page

File: components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js [hunk 2]
```diff
@@ -15,14 +19,46 @@ import StyledInputField from '../StyledInputField';
...
+  }
+  @media screen and (max-width: 40em) {
+    height: 192px;
+    width: 192px;
+  }
+  @media screen and (min-width: 40em) and (max-width: 52em) {
+    height: 208px;
+    width: 208px;
+  }
+`;
```
Comment (Betree) [inline components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js:59]: Just mentioning an alternative way to write that with `styled-system`'s [`css` helper ](https://styled-system.com/css) that may prevent typos in the media queries:

```suggestion
const Image = styled.img(
  css({
    width: [192, 208, 256],
    height: [192, 208, 256]
  })
);
```

File: components/accept-financial-contributions/SuccessPage.js [hunk 3]
```diff
@@ -115,18 +115,38 @@ class SuccessPage extends React.Component {
               <P fontSize="LeadCaption" lineHeight="Paragraph" mb={3} color="black.800">
                 <FormattedMessage
                   id="acceptContributions.tiers.paragraphOne"
-                  defaultMessage="We created a one-time donation tier for you to begin with. Go ahead and create different one-time and subscription tiers to define the levels or types of financial contributions your collective accepts."
+                  defaultMessage="We created a one-time donation tier for you to begin with. Go ahead and create different one-time and subscription tiers to define the levels or types of financial contributions your collective accepts. {knowMore}."
+                  values={{
+                    knowMore: (
+                      <SmallExternalLink href={TIERS_INFO_LINK} openInNewTab>
```
Comment (Betree) [inline components/accept-financial-contributions/SuccessPage.js:121]: **[:ice_cream: Details & personal preferences]** If the component always has `openInNewTab`, adding that to its definition may be simpler:

```es6
const SmallExternalLink = styled(StyledLink).attrs({ openInNewTab: true })`
  font-size: ${themeGet('fontSizes.LeadCaption')}px;
`;
```

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

File: components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js [hunk 4]
```diff
@@ -135,21 +193,95 @@ class AcceptContributionsMyself extends React.Component {
...
+                    addLoggedInUserAsAdmin
+                    excludeAdminFields
+                  />
+                ) : (
+                  <CreateNewOrg alignItems="center" onClick={() => this.setState({ miniForm: true })}>
+                    <PlusCircle size="24" color="gray" />
+                    <P fontSize="Caption" color="black.800" ml={2}>
+                      <FormattedMessage
+                        id="acceptContributions.organization.createNewOrganization"
+                        defaultMessage="Create new organization"
```
Comment (Betree) [inline components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js:277]: **[:sparkles: **Enhancement**]** We already have a string for this one. The `Organization` is title case, but according to https://docs.opencollective.com/help/about/terminology#capitalization that's how it should be.

```suggestion
                        id="Organization.CreateNew"
                        defaultMessage="Create new Organization"
```

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

File: components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js (hunk 2 above)
Comment (sbinlondon) [reply components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js:59]: Oh this is much better! Thanks 👍

Comment (sbinlondon) [reply components/accept-financial-contributions/AcceptContributionsMyselfOrOrg.js:62]: I tried it and it seems the size was not actually applying to the images. I will look further into it in the next PR.

**Pull request merged**